Assets/Scripts/Combatscripts/CombatStateController.cs

```diff
@@ -116,6 +116,18 @@ private void EndLevel() {
             // Debug.Log("currentEnemyCount: " + currentEnemyCount + "  currentFriendlyCount: " + currentFriendlyCount);
             // Debug.Log("GAME HAS ENEDED NOW");
             // Map
+
+            ScrapCollector scrapCollector = FindObjectOfType<ScrapCollector>();
```

Comment:
It isn't great to use FindObjectOfType for a bunch of reasons:
- It searches through every active object in the scene
- Generally very slow
- If there happen to be 2 objects of this type things could get strange

I would recommend making the scrapCollector a serialized variable

Reply:
Sounds good. Would I do something like this? 

[SerializeField] private ScrapCollector scrapCollector;

Reply:
Would I just drag and drop the scrapCollector in the Inspector after doing this?

Reply:
This definitely wasn't the best way of doing it, but I made ScrapController a variable in the GameStateManager script, dragged and dropped the ScrapController prefab into that script in the Inspector, and set the scrapController of CombatStateController inside GameStateManager using that. The reason I did this was because CombatStateController isn't attached to an empty GameObject in the scene anywhere. It's being called inside other scripts.